tests/kafkatest/tests/core/downgrade_test.py

```diff
@@ -81,6 +81,8 @@ def wait_until_rejoin(self):
                     timeout_sec=60, backoff_sec=1, err_msg="Replicas did not rejoin the ISR in a reasonable amount of time")
 
     @cluster(num_nodes=7)
+    @parametrize(version=str(LATEST_3_9), compression_types=["snappy"])
+    @parametrize(version=str(LATEST_3_9), compression_types=["zstd"], security_protocol="SASL_SSL")
```

Comment:
Do we not test with `none` and `static_membership` anymore?
I see that 3.8 also misses that parametrization. 3.7 has it, though.

Reply:
Perhaps we should use `matrix` to include those test suites.

```
    @matrix(version=[str(LATEST_2_5), str(LATEST_2_6), str(LATEST_2_7), str(LATEST_2_8), str(LATEST_3_0),
                     str(LATEST_3_1), str(LATEST_3_2), str(LATEST_3_3), str(LATEST_3_4), str(LATEST_3_5),
                     str(LATEST_3_6), str(LATEST_3_7), str(LATEST_3_8), str(LATEST_3_9)],
            compression_types=[["none"], ["zstd"], ["snappy"]],
            security_protocols=["SASL_SSL", PLAINTEXT],
            static_membership=[False, True]
    )
```

Reply:
Thank you for pointing that out! 
I made changes in this PR based on the 3.8 one, so I believe this was missed in 3.8 and unintentionally carried over here. 😬 
I’ll add it for both 3.8 and 3.9.

Reply:
With this new matrix, we run quite some more tests. I am not sure, if it was set the old way to NOT run so many tests.

Reply:
It will bring addition test for "PLAINTEXT" protocol ?

Reply:
As far as I understand the notation (I might be wrong since I am not too familiar with this), the old parametrization 
```
@parametrize(version=str(LATEST_3_7), compression_types=["snappy"])
@parametrize(version=str(LATEST_3_7), compression_types=["zstd"], security_protocol="SASL_SSL")
@matrix(version=[str(LATEST_3_7)], compression_types=[["none"]], static_membership=[False, True])
```
runs for each version:

snappy, plaintext, static_membership=False
zstd, sasl_ssl, static_membership=False
none, plaintext, static_membership=False
none, plaintext, static_membership=True

The new parametrization 
```
@matrix(version=[str(LATEST_2_5), str(LATEST_2_6), str(LATEST_2_7), str(LATEST_2_8), str(LATEST_3_0),
                     str(LATEST_3_1), str(LATEST_3_2), str(LATEST_3_3), str(LATEST_3_4), str(LATEST_3_5),
                     str(LATEST_3_6), str(LATEST_3_7), str(LATEST_3_8), str(LATEST_3_9)],
            compression_types=[["none"], ["zstd"], ["snappy"]],
            security_protocols=["SASL_SSL", "PLAINTEXT"],
            static_membership=[False, True]
  )
```
runs for each version:

snappy, plaintext, static_membership=False
snappy, plaintext, static_membership=True
snappy, sasl_ssl, static_membership=False
snappy, sasl_ssl, static_membership=True
zstd, plaintext, static_membership=False
zstd, plaintext, static_membership=True
zstd, sasl_ssl, static_membership=False
zstd, sasl_ssl, static_membership=True
none, plaintext, static_membership=False
none, plaintext, static_membership=True
none, sasl_ssl, static_membership=False
none, sasl_ssl, static_membership=True

This is three times more tests per version. I am wondering, if the intend of the old parametrization was to NOT run so many tests.

Reply:
> This is three times more tests per version. I am wondering, if the intend of the old parametrization was to NOT run so many tests.

Maybe we can remove snappy? or using the origin style is fine to me :)

Reply:
I think old style is more suitable for current situation.
I will update it later.

Reply:
> I think old style is more suitable for current situation.
I will update it later.

Yeah, I agree!